aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -211,12 +210,621 @@ Tensor& replication_pad1d_backward_out_cpu_template(
+  int64_t iheight;
+  int64_t iwidth;
+  int64_t oheight;
+  int64_t owidth;
+
+  if (input.dim() == 4)
+  {
+    nbatch = input.size(0);
+    dimw++;
+    dimh++;
+    dimslices++;
+  }
+
+  /* sizes */
+  nslices = input.size(dimslices);
+  iheight = input.size(dimh);
+  iwidth = input.size(dimw);
+  oheight = iheight + pad_t + pad_b;
+  owidth  = iwidth + pad_l + pad_r;
+
+  AT_CHECK(owidth == gradOutput_.size(dimw),
+      "gradOutput width unexpected. Expected: ", owidth, ", Got: ",
+      gradOutput_.size(dimw));
+  AT_CHECK(oheight == gradOutput_.size(dimh),
+      "gradOutput height unexpected. Expected: ", oheight, ", Got: ",
+      gradOutput_.size(dimh));
+
+  /* get contiguous gradOutput */
+  auto gradOutput = gradOutput_.contiguous();
+
+  /* resize */
+  gradInput.resize_as_(input);
+  gradInput.zero_();
+
+  /* backprop */
+  if (input.dim() == 3) {
+    AT_DISPATCH_FLOATING_TYPES(
+        input.type(), "replication_pad2d_backward", [&] {
+        replication_pad2d_backward_out_frame<scalar_t>(
+            gradInput.data<scalar_t>(),
+            gradOutput.data<scalar_t>(),
+            nslices,
+            iwidth, iheight,
+            owidth, oheight,
+            pad_l, pad_r,
+            pad_t, pad_b);
+        }
+        );
+  } else {
+    int64_t p;
+#pragma omp parallel for private(p)
+    for (p = 0; p < nbatch; p++) {
+      AT_DISPATCH_FLOATING_TYPES(
+          input.type(), "replication_pad2d_backward", [&] {
+          replication_pad2d_backward_out_frame<scalar_t>(
+              gradInput.data<scalar_t>() + p * nslices * iheight * iwidth,
+              gradOutput.data<scalar_t>() + p * nslices * oheight * owidth,
+              nslices,
+              iwidth, iheight,
+              owidth, oheight,
+              pad_l, pad_r,
+              pad_t, pad_b);
+          }
+          );
+    }
+  }
+  return gradInput;
+}
+
+static inline void shapeCheck(
+    const Tensor& input,
+    int pleft, int pright,
+    int ptop, int pbottom,
+    int pfront, int pback) {
+  int dimw = 3;
+  int dimh = 2;
+  int dimd = 1;
+  int dimslices = 0;
+  int64_t nslices;
+  int64_t idepth;
+  int64_t iheight;
+  int64_t iwidth;
+  int64_t odepth;
+  int64_t oheight;
+  int64_t owidth;
+
+  AT_CHECK(input.numel() > 0 && (input.dim() == 4 || input.dim() == 5),
+      "non-empty 4D or 5D (batch mode) tensor expected for input, but got: ", input);
+
+  if (input.dim() == 5)
+  {
+    dimw++;
+    dimh++;
+    dimd++;
+    dimslices++;
+  }
+
+  /* sizes */
+  nslices = input.size(dimslices);
+  idepth = input.size(dimd);
+  iheight = input.size(dimh);
+  iwidth = input.size(dimw);
+  odepth = idepth + pfront + pback;
+  oheight = iheight + ptop + pbottom;
+  owidth  = iwidth + pleft + pright;
+
+  AT_CHECK(owidth >= 1 || oheight >= 1 || odepth >= 1,
+      "input (D: ", idepth, " H: ", iheight, ", W: ", iwidth,
+      ") is too small."
+      " Calculated output D: ", odepth, " H: ", oheight, " W: ", owidth);
+
```

Comment:
What happened to the gradOutput check?

Reply:
I did blind port. in original implementation, gradOutput is always passed as NULL when THNN_(VolumetricReplicationPadding_shapeCheck) is called. So that part of code is not executed.